internal/templates/antiaffinity/template.go

```diff
@@ -0,0 +1,68 @@
+package antiaffinity
+
+import (
+	"fmt"
+
+	coreV1 "k8s.io/api/core/v1"
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"golang.stackrox.io/kube-linter/internal/check"
+	"golang.stackrox.io/kube-linter/internal/diagnostic"
+	"golang.stackrox.io/kube-linter/internal/extract"
+	"golang.stackrox.io/kube-linter/internal/lintcontext"
+	"golang.stackrox.io/kube-linter/internal/objectkinds"
+	"golang.stackrox.io/kube-linter/internal/templates"
+	"golang.stackrox.io/kube-linter/internal/templates/antiaffinity/internal/params"
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+func init() {
+	templates.Register(check.Template{
+		HumanName:   "Anti affinity not specified",
+		Key:         "anti-affinity",
+		Description: "Flag objects with multiple replicas but anti affinity not specified in the pod template spec",
+		SupportedObjectKinds: check.ObjectKindsDesc{
+			ObjectKinds: []string{objectkinds.DeploymentLike},
+		},
+		Parameters:             params.ParamDescs,
+		ParseAndValidateParams: params.ParseAndValidate,
+		Instantiate: params.WrapInstantiateFunc(func(p params.Params) (check.Func, error) {
+			return func(_ *lintcontext.LintContext, object lintcontext.Object) []diagnostic.Diagnostic {
+				replicas, found := extract.Replicas(object.K8sObject)
+				if !found {
+					return nil
+				}
+				if int(replicas) >= p.MinReplicas {
+					podTemplateSpec, hasPods := extract.PodTemplateSpec(object.K8sObject)
+					if !hasPods {
+						return nil
+					}
+					if affinity := podTemplateSpec.Spec.Affinity; affinity != nil && affinity.PodAntiAffinity != nil {
+						preferredAffinity := affinity.PodAntiAffinity.PreferredDuringSchedulingIgnoredDuringExecution
+						requiredAffinity := affinity.PodAntiAffinity.RequiredDuringSchedulingIgnoredDuringExecution
+						for _, preferred := range preferredAffinity {
+							if isAffinityTerm(preferred.PodAffinityTerm, podTemplateSpec) {
+								return nil
+							}
+						}
+						for _, required := range requiredAffinity {
+							if isAffinityTerm(required, podTemplateSpec) {
+								return nil
+							}
+						}
+					}
+					return []diagnostic.Diagnostic{{Message: fmt.Sprintf("object with (%v) replicas but anti affinity is not specified in pod spec ", replicas)}}
+				}
+				return nil
+			}, nil
+		}),
+	})
+}
+
+func isAffinityTerm(affinityTerm coreV1.PodAffinityTerm, podTemplateSpec coreV1.PodTemplateSpec) bool {
+	labelSelector, _ := metaV1.LabelSelectorAsSelector(affinityTerm.LabelSelector)
```

Comment:
You still need to check the error and `return false` if `err != nil`. Otherwise it will be `nil` and the next line will panic.